composer/trainer/trainer.py

```diff
@@ -3479,13 +3485,20 @@ def _eval_loop(
                                         )[0]
 
                             self.engine.run_event(Event.EVAL_BEFORE_FORWARD)
+                            if self.state.precision is Precision.AMP_FP8:
+                                import transformer_engine.pytorch as te
+                                log.info("Disabling AMP FP8 for eval loop")
+                                fp8_ctx = te.fp8_autocast(enabled=False)
+                            else: 
+                                fp8_ctx = contextlib.nullcontext()
```

Comment:
maybe make helper fn? there are other parts in codebase (eg DPO callback) that require a pass which would want this context